backend/test_observer/controllers/reports/reports.py

```diff
@@ -72,10 +107,89 @@ def get_testresults_report(
         .where(Artefact.created_at >= start_date, Artefact.created_at <= end_date)
     )
 
-    filename = "testresults_report.csv"
-    with open(filename, "w") as csvfile:
-        writer = csv.writer(csvfile)
-        writer.writerow(TESTRESULTS_REPORT_COLUMNS)
-        writer.writerows(cursor)
+    return _report_csv_format(
+        "testresults_report.csv",
+        TESTRESULTS_REPORT_COLUMNS,
+        cursor,
+    )
 
-    return filename
+
+def _get_test_execution_reports_query(
+    columns: list[InstrumentedAttribute],
+    include_test_events: bool,
+    start_date: datetime,
+    end_date: datetime,
+) -> Select:
+    """
+    Builds the query that retrieves the test executions based on the parameters set
+    """
+    latest_builds = queries.latest_artefact_builds.subquery()
+    base_query = (
+        select(*columns)
+        .join_from(TestExecution, Environment)
+        .join_from(TestExecution, latest_builds)
+        .join_from(latest_builds, Artefact)
+        .join_from(Artefact, Stage)
+        .join_from(Stage, Family)
+        .join(
+            ArtefactBuildEnvironmentReview,
+            (ArtefactBuildEnvironmentReview.artefact_build_id == latest_builds.c.id)
+            & (ArtefactBuildEnvironmentReview.environment_id == Environment.id),
+        )
+        .where(
+            TestExecution.created_at >= start_date, TestExecution.created_at <= end_date
+        )
+    )
+
+    if include_test_events:
+        return base_query.outerjoin(
+            TestEvent, TestExecution.id == TestEvent.test_execution_id
+        ).group_by(*TEST_EXECUTIONS_REPORT_COLUMNS)
+
+    return base_query
```

Comment:
if we don't have to require latest builds only then I assume we can simplify this query. I imagine we can have the TestExecution table query be at the top level that has a join on a subquery that aggregates the test events. That will likely be much simpler